src/jsonToMarkdown.tsx

```diff
@@ -0,0 +1,204 @@
+    return `${child}`
+  },
+  ul: (attrs: any, child: any) => {
+    return `${child}`
+  },
+  li: (attrs: any, child: any) => {
+    return `${child}`
+  },
+  a: (attrsJson: any, child: any) => {
+    return `[${child}](${attrsJson.url})`
+  },
+  hr: (attrs: any, child: any) => {
+    return `
+  
+----------`
+  },
+  span: (attrs: any, child: any) => {
+    return `${child}`
+  },
+  reference: (attrsJson: any, child: any): any  => {       
+    if(attrsJson?.['display-type'] === 'display') {
+      if(attrsJson) {
+        let assetName = attrsJson?.['asset-name'] ? attrsJson['asset-name'] : 'enter image description here'
+      let assetURL = attrsJson?.['asset-link'] ? attrsJson['asset-link'] : ''
+      return `
+      
+![${assetName}]
+(${assetURL})`
+      }
+    }
+    else if(attrsJson?.['display-type'] === 'link') {
+      if(attrsJson) {
+        return `[${child}](${attrsJson?.['href'] ? attrsJson['href'] : "#"})`
+      }
+    }
+  },
+  fragment: (attrs: any, child: any) => {
+    return child
+  },
+}
+const TEXT_WRAPPERS: IJsonToMarkdownTextTags = {
+  'bold': (child: any, value: any) => {
+    return `**${child}**`;
+  },
+  'italic': (child: any, value: any) => {
+    return `*${child}*`;
+  },
+  // 'underline': (child: any, value: any) => {
+  //   return `<u>${child}</u>`;
+  // }, underline is not supported in markdown
+  'strikethrough': (child: any, value: any) => {
+    return `~~${child}~~`;
+  },
+  'superscript': (child: any, value: any) => {
+    return `^${child}^`;
+  },
+  'subscript': (child: any, value: any) => {
+    return `~${child}~`;
+  },
+  'inlineCode': (child: any, value: any) => {
+    return `\`${child}\``
+  },
+}
+
+const getOLOrULStringFromJson = (value: any) => {
+  let child = ''
+  let nestedListFound = false
+  if(value.type === 'ol'){
+  let start = parseInt(value?.attrs?.start || 1)
+  Array.from(value.children).forEach((val: any, index) => { 
+    if(val.hasOwnProperty('type') && val.type === 'li' && value.children[index + 1] && value.children[index + 1]?.type === 'ol'){
+      let liChildren = jsonToMarkdownSerializer(val)
+      let nestedListChildren = getOLOrULStringFromJson(value.children[index + 1])
+      let indentedNestedListChildren = nestedListChildren.split('\n').filter((child) => child.length).map((child) => `  ${child}`).join('\n')
+      child += `${index + start}. ${liChildren}\n${indentedNestedListChildren}\n`
+      nestedListFound = true
+    }
+    else if(val.hasOwnProperty('type') && val.type !== 'ol'){
+      let liChildren = jsonToMarkdownSerializer(val) 
+      child += `${nestedListFound ? (index + start - 1): index + start}. ${liChildren}\n`
+    }
+
+  })
+  }
+  if(value.type === 'ul') {
+    let symbol = value?.attrs?.listStyleType || '- '
+    Array.from(value.children).forEach((val: any, index) => {
+      if(val.hasOwnProperty('type') && val.type === 'li' && value.children[index + 1] && value.children[index + 1]?.type === 'ol'){
```

Comment:
@Ashwini-Rathod There's a lot of duplicated code in the getOLOrULStringFromJson function. can we create a helper function to handle the common logic?